MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -8423,4 +8454,96 @@ public String getCommanderAddress() {
         return commanderRank;
     }
 
+    public int stockUpPartsInUse(Set<PartInUse> partsInUse)  {
+        int bought = 0;
+        for(PartInUse piu : partsInUse) {
+            int toBuy = findStockUpAmount(piu);
+            if(toBuy > 0) {
+                IAcquisitionWork partToBuy = piu.getPartToBuy();
+                getShoppingList().addShoppingItem(partToBuy, toBuy, this);
+            }
+            bought += 1;
+        }
+        return bought;
+    }
+
+    public void stockUpPartsInUseGM(Set<PartInUse> partsInUse)  {
+        for(PartInUse piu : partsInUse) {
+            int toBuy = findStockUpAmount(piu);
+            while(toBuy > 0) {
+                IAcquisitionWork partToBuy = piu.getPartToBuy();
+                getQuartermaster().addPart((Part) partToBuy.getNewEquipment(), 0);
+                -- toBuy;
+            }
+        }
+    }
+
+    private int findStockUpAmount(PartInUse piu) {
+        IAcquisitionWork partToBuy = piu.getPartToBuy();
+        int inventory = piu.getStoreCount() + piu.getTransferCount() + piu.getPlannedCount();
+        int needed = (int)Math.ceil(piu.getRequestedStock()/100.0 * piu.getUseCount());
+        int toBuy = needed-inventory;
+
+        if(piu.getIsBundle() == true) {
+            toBuy = (int)Math.ceil((float)toBuy * piu.getTonnagePerItem() / 5);
+            //special case for ammo, need to track down if there's a way to code this properly
+        }
+
+        if(toBuy > 0) {
+            System.out.println("TPI: " + piu.getTonnagePerItem() + " " + String.format("Inv: %d needed: %d tobuy: %d", inventory, needed, toBuy));
+            System.out.println("||");
+        }
+
+       
+        return toBuy;
+    }
+
+    //Simple getters and setters for our stock map
+    public Map<String,Double> getPiuStockMap() {
```

Comment:
You're probably fed up of seeing me write this by now, but we don't want abbreviated method names either. Especially for public methods.